proposals/static-lambdas.md

```diff
@@ -0,0 +1,32 @@
+# Static lambdas
+
+## Summary
+
+Support lambdas that disallow capturing state from the enclosing scope.
+
+## Motivation
+
+Avoid unintentionally capturing state from the enclosing context.
+
+## Detailed design
+
+A lambdas with `static` cannot capture state from the enclosing scope.
+As a result, locals, parameters, and `this` from the enclosing scope are not available within a `static` lambda.
+
+A `static` lambda cannot reference instance members from an implicit or explicit `this` or `base` reference.
+
+A `static` lambda may reference `static` members from the enclosing scope.
+
+A `static` lambda may reference `constant` definitions from the enclosing scope.
+
+`nameof()` in a `static` lambda may reference locals, parameters, or `this` or `base` from the enclosing scope.
+
+Accessibility rules for `private` members in the enclosing scope are the same for `static` and non-`static` lambdas.
+
+No guarantee is made as to whether a `static` lambda definition is emitted as a `static` method in metadata. This is left up to the compiler implementation to optimize.
+
```

Comment:
i agree with this.  Should we consider something additional like:

> Implementations are recommended to emit a static lambda with a singleton allocation for improved performance

?

Reply:
I'm fine to leave that as an implementation detail

Reply:
wouldn't that be useful with function pointers?
```cs
unsafe class Instance {
    void Use() {
        delgate*<Instance, string> f = &static i => i.ToString();
        f(this);
    }
}
```
In the proposal, the recommended workaround for instance methods is to use static local functions. I think a major advantage of using static lambdas over static local functions here is type inference and avoiding typing out the whole signature so it could be of help in this scenario as well.